File: Decorator/Python/Salad.py
# -*- coding: utf-8 -*-
"""
Created on Thu May 19 19:11:06 2022

@author: Mahmoud Saeed
"""

import SandwichDecorator

class Salad(SandwichDecorator.SandwichDecorator):
    
    def __init__(self , sandwich):
        super().__init__(sandwich)
        
    def getCost(self):
        return super().getCost() + 1.0 
    
    def getDesc(self):
        return super().getDesc() + " + Salad "

File: Decorator/Java/src/decoratorSandwich/salad.java
package decoratorSandwich;

public class salad extends SandwichDecorator {
    public salad(Sandwich sandwich) {
        super(sandwich);
    }

    @Override
    public String getDescription() {
        return super.getDescription() + "  salad";

    }

    @Override
    public double getcost() {
        return super.getcost()+0.5;
    }
}
